### PR TITLE
Describe configuration behaviour with multiple mailers

### PR DESCRIPTION
**Pull request by xelan**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | symfony/swiftmailer-bundle#111 |

This PR describes the behaviour when multiple mailers are configured.

### CODE REVIEW

File: reference/configuration/swiftmailer.rst [hunk 1]
```diff
@@ -303,3 +303,9 @@ Each mailer is registered as a service::
 
     // returns the second mailer
     $container->get('swiftmailer.mailer.second_mailer');
+
+.. caution::
+
+    The configuration in the ``swiftmailer`` key (see above) is a shortcut
+    syntax for the default mailer. If you use multiple mailers, you need to
+    set options like ``disable_delivery`` separately for the additional mailers.
```
Comment (xabbuh) [inline reference/configuration/swiftmailer.rst]: I would reword this a bit:

```rst
.. caution::

    When configuring multiple mailers, options must be placed under the appropriate
    mailer key of the configuration. Options placed directly under the `swiftmailer`
    key will not be applied to all mailers, but will configure a mailer named
    ``default`` instead.
```

What do you think?

**Comment by xabbuh** [comment]
Thanks @xelan for making the documentation better! I have left just one minor suggestion. What do you think about it?

File: reference/configuration/swiftmailer.rst (hunk 1 above)
Comment (xelan) [inline reference/configuration/swiftmailer.rst]: Thank you, Christian, I think your wording sounds great and explains the issue well.
If i interpret the [configuration code](https://github.com/symfony/swiftmailer-bundle/blob/master/DependencyInjection/SwiftmailerExtension.php) correctly, the default mailer doesn't have to be named `default`, but it has to be configured as `default_mailer`.

What do you think of the following suggestion?

```yaml
.. caution::

    When configuring multiple mailers, options must be placed under the appropriate
    mailer key of the configuration. Options placed directly under the ``swiftmailer``
    key will not be applied to all mailers, but will configure the ``default_mailer`` instead.
```

**Comment by xabbuh** [comment]
:+1:

**Comment by sstok** [comment]
:+1:

**Comment by xabbuh** [comment]
Hm, looking at the code again I think this only true if you do not change the default mailer using the `default_mailer` option (which defaults to `default`).

@stof What do you think?

**Comment by stof** [comment]
@xabbuh using the shorthand syntax in multiple files with a non-default name for the mailer indeed works only if you redefine the name in all files (the same is true for Doctrine shortcuts btw).
Anyway, the recommendation should be to never use the short syntax if you use multiple mailers (and to avoid changing the mailer name if you use only one, as you don't care about the name anyway in such case)

**Comment by pardery** [comment]
Thanks @stof, @xelan, and everyone else for this info/work.

**Comment by WouterJ** [comment]
Thanks @xelan for your contribution! I've merged your PR into the 2.3 branch. After the merge, I've changed the wording a bit in 34c0771 to not explain what happends if someone doesn't follow this recommendation (they should simply never do it).
